src/sentry/snuba/metrics/extraction.py

```diff
@@ -1162,7 +1166,11 @@ def _query_str_for_hash(self) -> str:
     @cached_property
     def query_hash(self) -> str:
         str_to_hash = self._query_str_for_hash
-        hash = hashlib.shake_128(bytes(str_to_hash, encoding="ascii")).hexdigest(4)
+        if self.spec_version.flags == {"use_new_unicode_and_env_logic"}:
+            hash = hashlib.shake_128(bytes(str_to_hash, encoding="utf-8")).hexdigest(4)
+        else:
+            hash = hashlib.shake_128(bytes(str_to_hash, encoding="ascii")).hexdigest(4)
+
```

Comment:
UTF-8 is a superset of ASCII, so you either get exactly the same hash for both or a hash from new one, encoding error from the other. You'll never get two hashes that are different. 

So what I'm basically saying there is not need to complicate so much with having to versions etc.  Changing `encoding="ascii"` to `encoding="utf-8"` is fully backwards compatible.

Reply:
That's really good to know. Thanks @vartec !